### PR TITLE
(RSC15j) Add `Host` header for fallback hosts.

### PR DESCRIPTION
**Pull request by lukasz-szyszkowski**

### CODE REVIEW

File: Source/ARTNSURL+ARTUtils.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+//
+
+#import <Foundation/Foundation.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface NSURL (ARTUtils)
+
++ (nullable NSURL *)URLWith:(NSURL *)url host:(NSString *)host;
+
```
Comment (maratal) [inline Source/ARTNSURL+ARTUtils.h:16]: I would go with `copyFromURL:withHost`

File: Source/ARTRest.m [hunk 2]
```diff
@@ -356,11 +357,12 @@ - (NSString *)description {
                 NSString *host = [blockFallbacks popFallbackHost];
                 if (host != nil) {
                     [self.logger debug:__FILE__ line:__LINE__ message:@"RS:%p host is down; retrying request at %@", self, host];
+                    
                     NSMutableURLRequest *newRequest = [request copy];
-                    NSURL *url = request.URL;
-                    NSString *urlStr = [NSString stringWithFormat:@"%@://%@:%@%@?%@", url.scheme, host, url.port, url.path, (url.query ? url.query : @"")];
-                    newRequest.URL = [NSURL URLWithString:urlStr];
```
Comment (maratal) [inline Source/ARTRest.m:362]: `urlString`

Comment (lukasz-szyszkowski) [reply Source/ARTRest.m:362]: that's removed part of the code, it wasn't mine

File: Source/ARTNSURL+ARTUtils.h (hunk 1 above)
Comment (lukasz-szyszkowski) [reply Source/ARTNSURL+ARTUtils.h:16]: changes here 64961aa88a99211919ffffd946ed37ad1203f4d2

**Review by maratal — APPROVED** [review]

**Pull request merged**